File: native-bridge/android/src/main/java/com/kaleyra/video_hybrid_native_bridge/events/Events.kt
// To parse the JSON, install kotlin's serialization plugin and do:
//
// val json   = Json { allowStructuredMapKeys = true }
// val events = json.parse(Events.serializer(), jsonString)

package com.kaleyra.video_hybrid_native_bridge.events

import kotlinx.serialization.*
import kotlinx.serialization.json.*
import kotlinx.serialization.descriptors.*
import kotlinx.serialization.encoding.*

/**
 * This enum defines all the events that may be handled
 * <br/>
 * <br/>
 * You can listen to these events via [[BandyerPlugin.on]]
 */
@Serializable
enum class Events(val value: String) {
    @SerialName("accessTokenRequest") AccessTokenRequest("accessTokenRequest"),
    @SerialName("callError") CallError("callError"),
    @SerialName("callModuleStatusChanged") CallModuleStatusChanged("callModuleStatusChanged"),
    @SerialName("chatError") ChatError("chatError"),
    @SerialName("chatModuleStatusChanged") ChatModuleStatusChanged("chatModuleStatusChanged"),
    @SerialName("iOSVoipPushTokenInvalidated") IOSVoipPushTokenInvalidated("iOSVoipPushTokenInvalidated"),
    @SerialName("iOSVoipPushTokenUpdated") IOSVoipPushTokenUpdated("iOSVoipPushTokenUpdated"),
    @SerialName("setupError") SetupError("setupError");
}

File: native-bridge/TypeScript/Events.ts
// Copyright © 2018-2023 Kaleyra S.p.a. All Rights Reserved.
// See LICENSE for licensing information

/**
 * This enum defines all the events that may be handled
 * <br/>
 * <br/>
 * You can listen to these events via [[BandyerPlugin.on]]
 */
export enum Events {
  /**
   * Used for access token retrieval
   */
  accessTokenRequest = 'accessTokenRequest',
  /**
   * Used to notify setup errors
   */
  setupError = 'setupError',
  /**
   * Used to notify call module status changed
   */
  callModuleStatusChanged = 'callModuleStatusChanged',
  /**
   * Used to notify call errors
   */
  callError = 'callError',
  /**
   * Used to notify chat errors
   */
  chatError = 'chatError',
  /**
   * Used to notify chat module status changed
   */
  chatModuleStatusChanged = 'chatModuleStatusChanged',
  /**
   * Used to notify voip push token updates
   */
  iOSVoipPushTokenUpdated = 'iOSVoipPushTokenUpdated',
  /**
   * Used to notify voip push token invalidation
   */
  iOSVoipPushTokenInvalidated = 'iOSVoipPushTokenInvalidated',
}
